compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -1477,32 +1468,22 @@ public void visit(BLangNumericLiteral literalExpr) {
 
     @Override
     public void visit(BLangTupleVarRef varRefExpr) {
-        varRefExpr.expressions.forEach(expression -> expression.accept(this));
-        if (varRefExpr.restParam != null) {
-            BLangExpression restExpr = (BLangExpression) varRefExpr.restParam;
-            restExpr.accept(this);
-        }
+        varRefExpr.expressions.forEach(this::acceptNode);
+        this.acceptNode((BLangNode) varRefExpr.restParam);
```

Comment:
We have https://github.com/ballerina-platform/ballerina-lang/issues/31658 to track covering these visits